src/addon/ESMX/Driver/CMakeLists.txt

```diff
@@ -415,53 +398,79 @@ foreach(CMP IN ITEMS ${COMPS})
     message(FATAL_ERROR "Invalid BUILD_TYPE: ${CMP_BUILD_TYPE} for ${CMP}")
   endif()
 
-  # include modules and link libraries
-  if(${CMP_LINK_INTO_APP} STREQUAL "True")
+  # look for CMake target and conditionally include config, modules, link libs
+  if(TARGET ${CMP_CMAKE_TARGET})
+    if(${CMP_LINK_INTO_APP} STREQUAL "True")
+      message(STATUS "ESMX Found CMake target dependency: ${CMP_CMAKE_TARGET}")
+      target_link_libraries(esmx_driver PUBLIC ${CMP_CMAKE_TARGET})
+    endif()
+  elseif(${CMP_BUILD_TYPE} STREQUAL "cmake")
+    message(STATUS "ESMX Found CMake subdirectory: ${CMP_SOURCE_DIR}")
+    is_subdirectory(${CMP_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR} IS_SUB)
+    if(IS_SUB)
+      # component source dir under current source dir
+      add_subdirectory(${CMP_SOURCE_DIR})
+    else()
+      # component source dir NOT under current source dir
+      add_subdirectory(${CMP_SOURCE_DIR} ${CMAKE_BINARY_DIR}/${CMP})
```

Comment:
Is it possible to always use this line whether or not the component is a subdirectory?

Reply:
Yes, that is the key CMake feature that allows us to integrate `ESMX` from its installation location into any other project's CMake build system. The only requirement (from CMake) is that when the provided source directory is not a subdir, then you must provide the binary directory of where CMake is supposed to put the build artifacts - since it doesn't know where to integrate it automatically. It's a great feature of CMake.

Reply:
I mean to clean this up to always use `add_subdirectory(${CMP_SOURCE_DIR} ${CMAKE_BINARY_DIR}/${CMP})` and remove `add_subdirectory(${CMP_SOURCE_DIR})`

Reply:
That would probably work, but seems better to just let CMake do its thing for real subdirs.

Reply:
After further discussion on the core call today, it was decided that simplifying the logic, and eliminating the need for the `is_subdirectory`  function is large enough a benefit to make the change. The change has been pushed.